### PR TITLE
Add app.vup.Vup

### PR DESCRIPTION
**Pull request by lukehmcc**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - "--filesystem=xdg-run/app/com.discordapp.Discord:create" is used to talk with discord to allow for status notification
  - "--filesystem=home" this is a file management type app so access to the home directory is required
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/redsolver/vup**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: app.vup.Vup.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+runtime: org.gnome.Platform
+runtime-version: '43'
+sdk: org.gnome.Sdk
+add-extensions:
+  org.freedesktop.Platform.ffmpeg-full:
+    directory : "lib/ffmpeg"
+    version: "21.08"
+    add-ld-path: .  
+cleanup-commands:
+- mkdir -p ${FLATPAK_DEST}/lib/ffmpeg
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:11]: should should be in a `post-install` somewhere. it's not a cleanup

File: app.vup.Vup.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- "--device=dri"
+- "--socket=pulseaudio"
+- "--share=network"
+- "--filesystem=home"
+- "--talk-name=org.kde.StatusNotifierWatcher"
+- "--talk-name=org.freedesktop.Notifications"
+- "--filesystem=xdg-run/app/com.discordapp.Discord:create"
+modules:
+- name: libjsoncpp
+  buildsystem: cmake
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:27]:
```suggestion
  buildsystem: cmake-ninja
```

File: app.vup.Vup.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  - type: archive
+    url: https://github.com/open-source-parsers/jsoncpp/archive/refs/tags/1.7.5.tar.gz
+    sha256: 4338c6cab8af8dee6cdfd54e6218bd0533785f552c6162bb083f8dd28bf8fbbe
+- name: libsodium
+  buildsystem: autotools
+  sources:
+  - type: archive
+    url: https://github.com/jedisct1/libsodium/archive/refs/tags/1.0.18-RELEASE.tar.gz
+    sha256: b7292dd1da67a049c8e78415cd498ec138d194cfdb302e716b08d26b80fecc10
+- name: intltool
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:41]: please use the shared-modules for this. https://github.com/flathub/shared-modules/

File: app.vup.Vup.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  - type: archive
+    url: https://gitlab.gnome.org/World/intltool/-/archive/release-0_51_0/intltool-release-0_51_0.tar.gz
+    sha256: 3d16e312265cf92658668d1aace692e3bfa8ff08b3449a124571b320ae3a8eac
+- shared-modules/libappindicator/libappindicator-gtk3-introspection-12.10.json
+- shared-modules/libsecret/libsecret.json
+- name: vup
+  buildsystem: simple
+  build-commands:
+  - mkdir -p Vup
+  - unzip Vup-0.12.0-BETA-Linux.zip -d build
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:53]: I'll assume this is a x86_64 specific binary. But there is still an aarch64 to be done.

File: app.vup.Vup.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  - cp -r build/* Vup
+  - cp -r Vup /app/
+  - chmod +x /app/Vup/vup
+  - mkdir -p /app/bin
+  - ln -s /app/Vup/vup /app/bin/vup
+  - mkdir -p /app/share/metainfo
+  - cp -r app.vup.Vup.metainfo.xml /app/share/metainfo/
+  - mkdir -p /app/share/icons/hicolor/scalable/apps
+  - cp -r icon.png /app/share/icons/hicolor/scalable/apps/app.vup.Vup.png
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:62]: a PNG is NOT scalable

File: app.vup.Vup.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  - cp -r app.vup.Vup.desktop /app/share/applications/
+  - mkdir -p /app/share/appdata
+  - cp -r app.vup.Vup.metainfo.xml /app/share/appdata/
+  sources:
+  - type: file
+    path: icon.png
+  - type: file
+    path: app.vup.Vup.metainfo.xml
+  - type: file
+    path: app.vup.Vup.desktop
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:73]: neither of these three file is found.

Shouldn't the come from upstream?

File: app.vup.Vup.yaml (hunk 4 above)
Comment (hfiguiere) [reply app.vup.Vup.yaml:53]: why not building from source?

File: app.vup.Vup.yaml [hunk 7]
```diff
@@ -0,0 +1,78 @@
+app-id: app.vup.Vup
+runtime: org.gnome.Platform
+runtime-version: '43'
+sdk: org.gnome.Sdk
+add-extensions:
+  org.freedesktop.Platform.ffmpeg-full:
+    directory : "lib/ffmpeg"
+    version: "21.08"
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:8]:
```suggestion
    version: "22.08"
```

Gnome 43 use 22.08

But do you really need the GNOME runtime?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
This won't build as is

File: app.vup.Vup.yaml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  sources: 
+  - type: archive
+    url: https://gitlab.gnome.org/World/intltool/-/archive/release-0_51_0/intltool-release-0_51_0.tar.gz
+    sha256: 3d16e312265cf92658668d1aace692e3bfa8ff08b3449a124571b320ae3a8eac
+- shared-modules/libappindicator/libappindicator-gtk3-introspection-12.10.json
+- shared-modules/libsecret/libsecret.json
+- name: vup
+  buildsystem: simple
+  build-commands:
+  - mkdir -p Vup
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:52]:
```suggestion
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: app.vup.Vup.yaml (hunk 4 above)
Comment (lukehmcc) [reply app.vup.Vup.yaml:53]: > why not building from source?

Would it be preferred to download the entirety of flutter to then build this project? It seems much simpler to just build from a release file.

Comment (hfiguiere) [reply app.vup.Vup.yaml:53]: from a release for a single architecture.

But then you mentionned Flutter...

Comment (hfiguiere) [reply app.vup.Vup.yaml:53]: the problem here is that the aarch64 flatpak is built with the x86_64 binaries.

Comment (lukehmcc) [reply app.vup.Vup.yaml:53]: So my only real option here would be to have a precompiled arrch64 image on the github releases page. Dart does not support cross compilation. Other flutter apps seemed to not have included aarch64 builds at all: [pstube](https://github.com/flathub/io.github.prateekmedia.pstube/blob/master/io.github.prateekmedia.pstube.yml), [spottube](https://github.com/flathub/com.github.KRTirtho.Spotube/blob/master/com.github.KRTirtho.Spotube.yml).

File: app.vup.Vup.yaml (hunk 7 above)
Comment (lukehmcc) [reply app.vup.Vup.yaml:8]: I do not, I added it to get libsecret out of the box but it seems I can easily get that as a submodule so I'll switch it to freedesktop.

**Comment by lukehmcc** [comment]
One thing to note about this commit, I was trying to make sure that it compiled on my end (which it did) so I added the git submodule for `shared-modules`. If this is a problem let me know.

The only other issue I now see is the lack of aarch64 support.

File: app.vup.Vup.yaml (hunk 4 above)
Comment (hfiguiere) [reply app.vup.Vup.yaml:45]: this is still not solved.

File: app.vup.Vup.yaml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  sources:
+  - type: file
+    path: icon.svg
+  - type: file
+    path: app.vup.Vup.metainfo.xml
+  - type: file
+    path: app.vup.Vup.desktop
+  - type: file
+    only-arches:
+          - x86_64
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:66]: this doesn't do what you think it does.

File: app.vup.Vup.yaml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  - ln -s /app/Vup/vup /app/bin/vup
+  - mkdir -p /app/share/icons/hicolor/scalable/apps
+  - cp -r icon.svg /app/share/icons/hicolor/scalable/apps/app.vup.Vup.svg
+  - install -Dm644 app.vup.Vup.desktop -t /app/share/applications/
+  - install -Dm644 app.vup.Vup.metainfo.xml -t /app/share/metainfo/
+  post-install:
+  - "mkdir -p ${FLATPAK_DEST}/lib/ffmpeg"
+  sources:
+  - type: file
+    path: icon.svg
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:59]: icons is not square

File: app.vup.Vup.yaml [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+  buildsystem: simple
+  build-commands:
+  - unzip Vup-0.12.0-BETA-Linux.zip -d build
+  - install -d /app/Vup
+  - cp -r build/* /app/Vup
+  - chmod +x /app/Vup/vup
+  - mkdir -p /app/bin
+  - ln -s /app/Vup/vup /app/bin/vup
+  - mkdir -p /app/share/icons/hicolor/scalable/apps
+  - cp -r icon.svg /app/share/icons/hicolor/scalable/apps/app.vup.Vup.svg
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:52]:
```suggestion
  - install -Dm644 icon.svg /app/share/icons/hicolor/scalable/apps/app.vup.Vup.svg
```

Like the other suggestions.

**Comment by hfiguiere** [comment]
bot, build app.vup.Vup

File: app.vup.Vup.yaml [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+- "--share=network"
+- "--filesystem=home"
+- "--talk-name=org.kde.StatusNotifierWatcher"
+- "--talk-name=org.freedesktop.Notifications"
+- "--filesystem=xdg-run/app/com.discordapp.Discord:create"
+modules:
+- name: libjsoncpp
+  buildsystem: cmake-ninja
+  config_opts:
+  - -DCMAKE_BUILD_TYPE=release
```
Comment (hfiguiere) [inline app.vup.Vup.yaml:27]:
```suggestion
  - -DCMAKE_BUILD_TYPE=Release
```

File: app.vup.Vup.yaml (hunk 4 above)
Comment (lukehmcc) [reply app.vup.Vup.yaml:45]: Is having an aarch64 build required? 

If not, then if I were to build an aarch64 binary separately and put it in the main apps release page and then used it for aarch64 flatpak builds would that solve this issue?

File: app.vup.Vup.yaml (hunk 10 above)
Comment (lukehmcc) [reply app.vup.Vup.yaml:59]: It is square. In the svg file: `viewBox="-11.113125 0 201.89867 201.89867"`
So the viewbox is `201.89867 x 201.89867`

File: app.vup.Vup.yaml (hunk 9 above)
Comment (lukehmcc) [reply app.vup.Vup.yaml:66]: I looked at the [command reference](https://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html). I get it now.

File: app.vup.Vup.yaml (hunk 10 above)
Comment (hfiguiere) [reply app.vup.Vup.yaml:59]: Then github is displaying it wrong

image

appstream will tell us anyway

File: app.vup.Vup.yaml (hunk 4 above)
Comment (hfiguiere) [reply app.vup.Vup.yaml:45]: It's in the documentation you acknowledged having read, and also the project you linked as examples:
https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

Comment (lukehmcc) [reply app.vup.Vup.yaml:45]: alright this should be resolved
I apologize for missing this

**Comment by lukehmcc** [comment]
All current issues should be resolved.

**Comment by hfiguiere** [comment]
bot, build app.vup.Vup

**Comment by hfiguiere** [comment]
Also since you are upstream, in the future, the icon, the appstream (metainfo) file and the desktop file should come in your tarball and source tree. Neither are specific to flathub, and the lack of icon and desktop file to to be considered as a bug.

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/app.vup.Vup.appdata.xml: FAILED:
• tag-invalid           : <metadata_license> is not valid [CCO-1.0]
• file-invalid          : <screenshot> url is a zero length file [https://github.com/vup-app/app.vup.Vup/blob/main/screenshots/computer.png?raw=true]
Validation of files failed
```

(the screenshot error is, I think, an infra problem)

File: app.vup.Vup.metainfo.xml [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    Generator for metainfo & .desktop files: 
+    https://www.freedesktop.org/software/appstream/metainfocreator/#/
+-->
+<component type="desktop-application">
+  <id>app.vup.Vup</id>
+  <name>Vup</name>
+  <summary>Vup is an open-source native E2EE cloud storage app for Skynet</summary>
+  <developer_name>redsolver</developer_name>
+  <url type="homepage">https://github.com/redsolver/vup</url>
+  <metadata_license>CCO-1.0</metadata_license>
```
Comment (hfiguiere) [inline app.vup.Vup.metainfo.xml:12]:
```suggestion
  <metadata_license>CC0-1.0</metadata_license>
```

Comment (hfiguiere) [reply app.vup.Vup.metainfo.xml:12]: it's a ZERO, not a O

Comment (lukehmcc) [reply app.vup.Vup.metainfo.xml:12]: oh dear that is quite the silly error

**Comment by barthalion** [comment]
bot, build app.vup.Vup

**Comment by barthalion** [comment]
The screenshot should be https://raw.githubusercontent.com/vup-app/app.vup.Vup/main/screenshots/computer.png but yeah, looks like there's something wrong on our end too.

**Comment by hfiguiere** [comment]
actually it shouldn't be `main` either because `main` (as a branch) will change. but that's not the problem at the moment.

**Comment by hfiguiere** [comment]
also note that the download link offered by github is still `https://github.com/vup-app/app.vup.Vup/raw/main/screenshots/computer.png` so it's really github here.

**Comment by hfiguiere** [comment]
About the screenshot error:

https://github.com/flathub/org.flatpak.Builder/issues/108

**Comment by hfiguiere** [comment]
bot, build app.vup.Vup

**Comment by lukehmcc** [comment]
> /merge @redsolver

Does redsolver have to do something to merge this? Or are you just pinging them?

**Comment by barthalion** [comment]
There was an error in the merge action. I have invited you manually. This is what the bot would usually put here:

>A repository for this submission has been created: https://github.com/flathub/org.kde.telly-skout
>
>You will receive an invitation to be a collaborator which will grant you write access to the repository above. The invite can be also viewed [here](https://github.com/flathub/org.kde.telly-skout/invitations).
>
>If you have never maintained an application before, common questions are answered in [the app maintenance guide](https://github.com/flathub/flathub/wiki/App-Maintenance).
>
>Thanks!

**Pull request closed without merging**